### PR TITLE
move shfmt execution from the super linter to pre-commit

### PR DESCRIPTION
**Pull request by altendky**
### Purpose:

I have no idea what shell formatter we should use.  But, https://github.com/Chia-Network/chia-blockchain/pull/17906 was updating the super linter and it started complaining a lot and I really dislike having formatting tools executed in the super linter since that doesn't integrate into our local checks and thus people have to manually apply the super linter results to their local copy.  So, this moves the super linter choice of shfmt into our pre-commit setup which runs both locally and in CI.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -33,6 +33,11 @@ repos:
     hooks:
       - id: prettier
         types_or: [ini, json, toml, yaml]
+  - repo: https://github.com/scop/pre-commit-shfmt
+    rev: v3.8.0-1
+    hooks:
+      - id: shfmt
+        args: ["-i", "2"]
```
Comment (arvidn) [inline .pre-commit-config.yaml:40]: I would think you want `--diff` as well. according to the [docs](https://github.com/mvdan/sh/blob/master/cmd/shfmt/shfmt.1.scd#generic-flags):

```
-d, --diff Error with a diff when the formatting differs.
```

Comment (altendky) [reply .pre-commit-config.yaml]: And `--write` apparently as well.  Thanks.  d464fb3781a2b688517cc5148d4b52dfc0aa95bc

**Review by arvidn — APPROVED** [review]
I suppose the super linter wasn't applying this formatting to these files

**Comment by altendky** [comment]
I added some explanation in the OP.  This change was inspired by a dependabot update of the super linter.

**Pull request merged**